Entities/Industry/Fireplace/FireplaceCommon.as

```diff
@@ -0,0 +1,17 @@
+﻿void IgniteFireplace(CBlob@ this)
+{
+	if (this.getSprite().isAnimation("fire")) return;
```

Comment:
Add a null check for the sprite just to be safe. And get the sprite once instead of getting it multiple times in this function.